source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -108,8 +121,16 @@ Http::FilterHeadersStatus CacheFilter::encodeHeaders(Http::ResponseHeaderMap& he
     insert_ = cache_.makeInsertContext(std::move(lookup_), *encoder_callbacks_);
     // Add metadata associated with the cached response. Right now this is only response_time;
     const ResponseMetadata metadata = {time_source_.systemTime()};
+    // TODO (capoferro): Note that there is currently no way to communicate back to the CacheFilter
+    // that an insert has failed. This is something we want to fix in future.
     insert_->insertHeaders(headers, metadata, end_stream);
+    if (end_stream) {
+      insert_status_ = InsertStatus::InsertSucceeded;
+    }
```

Comment:
should we be setting insert_status_ in an else here? Or is whatever was previously there known to be correct? Add comment if that's the case.

Reply:
All of the conditions in the `if` indicate reasons the response should not be cached, so I think this is correct? Maybe I'm missing something?

Reply:
oh, rereading your questions, I think I understand better... you mean there is not an `else` and maybe there should be. If it's not the end of the stream, we're still waiting for more insert activity to determine the `insert_status_`. At this point, `insert_status_` should be `absl::nullopt`. I suppose we could set it to an "still in progress" sort of status in an `else` here, if that's better.

Reply:
That, or at least a comment saying it will remain in a nullopt state.

As a reader it raises questions if *almost* all paths set a member variable.